### PR TITLE
Errors for unsorted data in mean algorithm 

### LINKED ISSUE #20658: Mean with unsorted data
bug report from James Lord: 

> 
> We discovered a feature: If we have two point workspaces with the same X values in a different order, for example from successive scans up and down in field, Mean doesn't notice the mismatch of X and happily goes ahead and generates junk. It ought at least to warn that the values don't match. The fix in this case is to use SortXAxis.
> 
> These data were from ALC analysis but we'd have the same with a results table from a sequential fit in Muon Analysis and then Convert to Workspace.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The mean algorithm calculates the average y value for each x value in the workspaces. If the workspaces did not have the same order for the data then this would produce the wrong result. This is because it simply iterated over the x values.

**To test:**
With the script below you can test:
1. It works if the data is good
2. If throws an error if one workspace has more spectra than the other
3. It throws an error if the x values are different between the two spectra 
```

# Create two  2-spectrum workspaces with Y values 1->8 & 3->10
dataX = [0,1,2,3,4,
         0,1,2,3,4]
dataY = [1,2,3,4,
         5,6,7,8]
ws_1 = CreateWorkspace(dataX, dataY, NSpec=2)


dataY = [3,4,5,6,
         7,8,9,10]
dataX = [0,1,2,3,4,
         0,1,3,2,4]
ws_2 = CreateWorkspace(dataX, dataY, NSpec=2)

result = Mean("ws_1, ws_2") # note the comma-separate strings
print "Mean of y values in first spectrum:", result.readY(0)
print "Mean of y values in second spectrum:", result.readY(1)
```

Fixes #20658. 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
need to add a test

File: Framework/PythonInterface/test/python/plugins/algorithms/MeanTest.py [hunk 1]
```diff
@@ -33,9 +35,23 @@ def test_throws_if_workspace_axis1_unequal(self):
...
             self.fail("Should not have got here. Should throw as axis1 is unequal in size between a and b.")
+        finally:
+            DeleteWorkspace(a)
+            DeleteWorkspace(b)
+
+    def test_throws_if_workspace_unorded(self):
+        a = CreateWorkspace(DataX=[1,2,1,2],DataY=[1,2,3,4],DataE=[1,1,1,1],UnitX='TOF',NSpec=2)
+        b = CreateWorkspace(DataX=[1,2,2,1],DataY=[1,2,3,4],DataE=[1,1,1,1],UnitX='TOF',NSpec=2)
+        try:
+            c = Mean(Workspaces='a,b') # 'a' and 'b' are different sizes.
```
Comment (gemmaguest) [inline Framework/PythonInterface/test/python/plugins/algorithms/MeanTest.py]: I think the comment here is incorrect (copy-paste error?)

**Review by gemmaguest — CHANGES_REQUESTED** [review]
The changes look good :+1:. I get the correct results if the data are ok, or sensible error messages if the workspaces need to be sorted or are not the same shape.

Just one minor point about a misleading comment.

**Review by gemmaguest — APPROVED** [review]

**Comment by gemmaguest** [comment]
All looks good now :+1:. The tests probably don't need to be re-run because the last commit was just a comment change.

**Pull request merged**